pkg/cli/admin/release/extract.go

```diff
@@ -359,9 +381,9 @@ func (o *ExtractOptions) Run(ctx context.Context) error {
 			context := "connected cluster"
 			inclusionConfig := manifestInclusionConfiguration{}
 			if o.InstallConfig == "" {
-				inclusionConfig, err = findClusterIncludeConfig(ctx, o.RESTConfig)
+				inclusionConfig, err = findClusterIncludeConfig(ctx, o.RESTConfig, versionInImageConfig)
 			} else {
-				inclusionConfig, err = findClusterIncludeConfigFromInstallConfig(ctx, o.InstallConfig)
+				inclusionConfig, err = findClusterIncludeConfigFromInstallConfig(ctx, o.InstallConfig, versionInImageConfig)
```

Comment:
Maybe I am missing something but I don't understand when is `versionInImageConfig` actually populated. What I'm reading is:

1. It is empty on L354 when declared
2. It can be populated on L371 inside the `opts.ImageConfigCallback` callback (which itself is populated on L355)
3. It can only be _actually_ populated once `opts.ImageConfigCallback` is called at least one
4. The callback is called in `pkg/cli/image/extract/extract.go:L426` in `o.Run()`
5. Here on L384 / L386 we expect this to be a populated string already but I do not see any opportunity betwen L372 and the L383 `if` where `o.Run()` could be called,  even indirectly.

How is `versionInImageConfig` not always an empty string on L384/L386??

Reply:
This is reiterated https://github.com/openshift/oc/pull/1958#discussion_r2089056410 but I do not understand the answer.